web/src/features/prompts/server/actions/getPromptsMeta.ts

```diff
@@ -0,0 +1,89 @@
+import { type GetPromptsMetaType } from "@/src/features/prompts/server/utils/validation";
+import { Prisma, prisma } from "@langfuse/shared/src/db";
+
+export type GetPromptsMetaParams = GetPromptsMetaType & { projectId: string };
+
+export const getPromptsMeta = async (
+  params: GetPromptsMetaParams,
+): Promise<PromptsMetaResponse> => {
+  const { projectId, page, limit } = params;
+
+  const promptsMeta = (await prisma.$queryRaw`
+    SELECT
+        p.name AS "promptName",
+        p.tags AS tags,
+        array_agg(DISTINCT p.version) AS versions,
+        array_agg(DISTINCT label) AS labels
+    FROM
+        prompts p
+    LEFT JOIN LATERAL unnest(p.labels) AS label ON true
+    WHERE 
+        p."project_id" = ${projectId} 
+        ${getPromptsFilterCondition(params)}
+    GROUP BY
+        p.name, p.tags --- tags are the same for all versions of a prompt
+    ORDER BY
+        p.name --- necessary for consistent pagination
+    LIMIT
+        ${limit}
+    OFFSET
+        ${limit * (page - 1)}
+  `) as PromptsMeta[];
+
+  const [{ count: totalItemsCount }] = (await prisma.$queryRaw`
+    SELECT COUNT(DISTINCT p.name) AS count
+    FROM prompts p
+    WHERE "project_id" = ${projectId} 
+    ${getPromptsFilterCondition(params)}
+  `) as { count: BigInt }[];
+
+  const totalItems = Number(totalItemsCount);
+  const totalPages = Math.ceil(totalItems / limit);
+
+  return {
+    data: promptsMeta,
+    pagination: { page, limit, totalPages, totalItems },
+  };
+};
+
+type PromptsMeta = {
+  promptName: string;
+  versions: number[];
+  labels: string[];
+  tags: string[];
+};
+
+export type PromptsMetaResponse = {
+  data: PromptsMeta[];
+  pagination: {
+    page: number;
+    limit: number;
+    totalPages: number;
+    totalItems: number;
+  };
+};
+
+const getPromptsFilterCondition = (params: GetPromptsMetaType) => {
```

Comment:
can we re-use the filters for the tables here? If yes, they should be found in the trpc route to fetch a list of prompts. I would preferthat function as it is way more powerful and tested against injectinos.

Reply:
```  
const filterCondition = tableColumnsToSqlFilterAndPrefix(
        input.filter,
        promptsTableCols,
        "prompts",
      );
```